utils.sh

```diff
@@ -117,11 +117,22 @@ function extract_and_read_artifact_version() {
 
 function read_artifact_version() {
     local MANIFEST_PATH
-    MANIFEST_PATH=$1
+    MANIFEST_PATH="$1"
 
     grep 'Implementation-Version' "$MANIFEST_PATH" | cut -d ' ' -f2 | sed -e 's/^[[:space:]]*//' -e 's/[[:space:]]*$//'
 }
 
+function is_snapshot_version() {
+    local VERSION
+    VERSION="$1"
+
+    if [[ "$VERSION" == *-SNAPSHOT ]]; then
+        return 0
```

Comment:
you return `0` and `1` but evaluate as a boolean `if ! is_snapshot_version "$NEW_VERSION"`

Reply:
Yes, this is correct. Function return some exit code using `return` and then `if command; then` or `if ! command; then` check if command returned success exit code (0) - true or not-success (!=0) - false

Reply:
Simplifed to 
```
function is_snapshot_version() {
    local VERSION
    VERSION="$1"

    [[ "$VERSION" == *-SNAPSHOT ]]
}
```

Reply:
and you do need a `return` statement in this simplified version? Confusing

Reply:
No need in simplified version. This is also possible variant of return exit code.